src/DIPS.Xamarin.UI/Controls/Toast/ToastCore.cs

```diff
@@ -9,23 +9,36 @@
 
 namespace DIPS.Xamarin.UI.Controls.Toast
 {
-    internal class ToastCore
+    internal class ToastCore : IDisposable
     {
         public ToastCore()
         {
-            Application.Current.PageAppearing -= OnPageAppearing;
-            Application.Current.PageAppearing += OnPageAppearing;
+            if (Application.Current == null)
+            {
+                return;
+            }
 
-            Application.Current.PageDisappearing -= OnPageDisappearing;
+            Application.Current.PageAppearing += OnPageAppearing;
             Application.Current.PageDisappearing += OnPageDisappearing;
         }
 
         private CancellationTokenSource CancellationSource { get; set; } = new CancellationTokenSource();
         private Dictionary<string, Grid> ToastContainers { get; } = new Dictionary<string, Grid>();
 
-        private void OnPageAppearing(object sender, Page e)
+        public void Dispose()
```

Comment:
Does this need to be public?

Reply:
![image](https://user-images.githubusercontent.com/8718745/89861102-9a794a80-dbc2-11ea-84b9-c3cd1eff162c.png)

Reply:
yes